src/components/MainSection.jsx

```diff
@@ -0,0 +1,111 @@
+import React, { Component, createRef } from 'react';
+import './MainSection.css';
+
+class MainSection extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      items: [],
+      visibleItems: 6,
+      allItemsLoaded: false,
+    };
+    this.quantityRefs = {};
+  }
+
+  componentDidMount() {
+    fetch('https://65de35f3dccfcd562f5691bb.mockapi.io/api/v1/meals')
+      .then((response) => response.json())
+      .then((data) => {
+        this.setState({ items: data });
+      })
+      .catch((error) => {
+        console.error("Error fetching the menu:", error);
+      });
+  }
+
+  handleSeeMore = () => {
+    const { visibleItems, items } = this.state;
+    const newVisibleCount = visibleItems + 6;
+
+    this.setState({
+      visibleItems: newVisibleCount,
+      allItemsLoaded: newVisibleCount >= items.length,
+    });
+  };
+
+  handleAddToCart = (itemId) => {
+    const ref = this.quantityRefs[itemId];
+    if (ref && ref.current) {
+      const quantity = Number(ref.current.value || 1);
+      if (quantity > 0) {
+        this.props.incrementItemCount(quantity);
+      }
+    } else {
+      console.error("Quantity input ref is null or undefined for item ID:", itemId);
```

Comment:
Please try to implement logic without undefined error.
You can disable button if there are no value, also you can put limit to input value = 1